xbmc/cdrip/EncoderFFmpeg.h

```diff
@@ -33,35 +33,48 @@ class CEncoderFFmpeg : public CEncoder
   bool Close() override;
 
 private:
+  struct EncoderException : public std::exception
```

Comment:
We do have the `Exception` class available here if you don't want to create something new https://github.com/xbmc/xbmc/blob/297b5f844a1db09be4118039e4c29a08e985fb56/xbmc/commons/Exception.h#L44

Reply:
Thanks for the reference to this, had completely forgotten that we have this :+1:.

Just looked over it, think to keep it at this point for now.

Once that the new handler refers purely to his place. "XbmcCommons::Exception" runs a lot through macros and try to avoid such uses if possible.

I think `XbmcCommons::Exception` would be a good new project to revise in the near future and make it conform to the latest C++ standards. Then I think we could integrate it here as well.

The only thing I'm going to do is integrate the string format into the "EncoderException" constructor and avoid the "StringUtils::Format" in the calls.